### PR TITLE
Feat(eos_designs): Set static-routes on node-type l3_interfaces

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Set static-routes on node-type l3_interfaces

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Changing key for `set_default_route` to `dhcp_accept_default_route`
- Adding key `static_routes: <list[str]>`

## How to test

Updated existing molecule tests with no config changes.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/docs/tables/wan-interfaces-settings.md [hunk 1]
```diff
@@ -157,6 +164,12 @@
             # The peer device IPv4 address (no mask). Used as default route gateway if `set_default_route` is true and `ip` is an IP address.
             peer_ip: <str>
 
+            # Configure IPv4 static routes pointing to `peer_ip`.
+            static_routes: # >=1 items
+
+                # IPv4_network/Mask
+              - <str>
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/tables/wan-interfaces-settings.md]: would we wan this to be list of dict instead to add some options?

```
edge1B(config)#ip route 1.1.1.1/32 ?
  A.B.C.D                    IPv4 address of the nexthop router
  A:B:C:D:E:F:G:H            IPv6 address of the nexthop router
  Dps                        DPS Tunnel Interface
  Ethernet                   Ethernet interface
  Fabric                     Fabric Interfaces
  Loopback                   Hardware interface used for looping packets
  Management                 Management interface
  Null0                      Interface that drops all traffic
  Port-Channel               Port-Channel Sub Interface
  Tunnel                     Tunnel interface
  Vlan                       Logical interface into a VLAN
  Vxlan                      VXLAN Tunnel Interface
  egress-vrf                 egress VRF for this route
  fe80::A:B:C:D%<interface>  IPv6 linklocal address of the nexthop router
  vtep                       Configure VXLAN Tunnel End Points
```

```
edge1B(config)#ip route 1.1.1.1/32 2.2.2.2 ?
  community  BGP communities
  metric     Metric for this route
  name       Next hop name
  tag        Route tag
  track      Track this route
  <1-255>    Administrative distance for this route
  <cr>
```

I can see this being intresting at some point

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/docs/tables/wan-interfaces-settings.md]: done

**Review by gmuloc — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type_l3_interfaces.schema.yml [hunk 2]
```diff
@@ -33,14 +33,10 @@ $defs:
...
-        set_default_route:
+        dhcp_accept_default_route:
           type: bool
           default: false
-          description: |-
-            Insert a default route to the `peer_ip` if `ip` is an ip address
-            or configure to accept a default route from DHCP if `ip` is `dhcp`.
-
-            AVD will error out if set to true, `ip` is an ip address and `peer_ip` is missing.
+          description: Accept a default route from DHCP if `ip` is `dhcp`.
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_node_type_l3_interfaces.schema.yml]:
```suggestion
          description: Accept a default route from DHCP if `ip_address` is set to `dhcp`.
```

**Review by carlbuchmann — APPROVED** [review]
LGTM, small suggestion on the description.
Tested in POC lab.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/static_routes.py [hunk 3]
```diff
@@ -22,43 +22,28 @@ def static_routes(self) -> list[dict] | None:
...
-            }
-
-            if static_route not in static_routes:
-                static_routes.append(static_route)
+            static_routes.extend(
+                {"destination_address_prefix": l3_interface_static_route["prefix"], "gateway": gateway}
+                for l3_interface_static_route in l3_interface_static_routes
+            )
 
         if static_routes:
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/static_routes.py:48]: return static_routes if static_routes else None

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/static_routes.py:48]: Even better to just `return static_routes or None`, but we have this pattern everywhere and we will fix it throughout at some point.

**Review by MaheshGSLAB — APPROVED** [review]

**Review by gmuloc — APPROVED** [review]
LGTM again

**Pull request merged**